sympy/solvers/ode/tests/test_systems.py

```diff
@@ -1071,8 +1071,8 @@ def test_sysode_linear_neq_order1_type2():
     eqs6 = [Eq(Derivative(f(x), x), -9*f(x) - 4*g(x)),
             Eq(Derivative(g(x), x), -4*g(x)),
             Eq(Derivative(h(x), x), h(x) + exp(x))]
-    sol6 = [Eq(f(x), C1*exp(-4*x)*Rational(-4, 5) + C2*exp(-9*x)),
-            Eq(g(x), C1*exp(-4*x)),
+    sol6 = [Eq(f(x), C2*exp(-4*x)*Rational(-4, 5) + C1*exp(-9*x)),
+            Eq(g(x), C2*exp(-4*x)),
```

Comment:
I can't work out why this has changed.

Reply:
Puzzled me, too. I think it has to do with the ordering of symbols that are used to replace the objects. `solve` takes care to make the "symbols" canonical before assigning dummies, but the `systems` code did not:
```python
rep = {func_deriv: Dummy() for func_deriv in funcs_deriv}  # ordered(funcs_deriv)
```
Just a hunch.

Reply:
The code that does this calls `linear_eq_to_matrix` but if I read this correctly it should preserve the order of `eqs` and `syms` so the output from `linear_eq_to_matrix` should be unchanged:
https://github.com/sympy/sympy/blob/39de9a2698ad4bb90681c0fdb70b30a78233145f/sympy/solvers/ode/systems.py#L471-L484

Reply:
I now suspect that it has to do with not returning a 0 coefficient from `_lin_eq2dict`. Here are equations being renumbered during the test involving `sol6`:
master 
```python
[Eq(f(x), (-4*_Dummy_148/5 - 4*Integral(0, x)/5)*exp(-4*x) + (_Dummy_147 + Integral(0, x))*exp(-9*x)), Eq(g(x), (_Dummy_148 + Integral(0, x))*exp(-4*x)), Eq(h(x), (_Dummy_149 + Integral(1, x))*exp(x))]
[Eq(f(x), (-4*C1/5 - 4*Integral(0, x)/5)*exp(-4*x) + (C2 + Integral(0, x))*exp(-9*x)), Eq(g(x), (C1 + Integral(0, x))*exp(-4*x)), Eq(h(x), (C3 + Integral(1, x))*exp(x))]
```
PR
```python
[Eq(f(x), _Dummy_121*exp(-9*x) - 4*_Dummy_122*exp(-4*x)/5), Eq(g(x), _Dummy_122*exp(-4*x)), Eq(h(x), (_Dummy_123 + Integral(1, x))*exp(x))]
[Eq(f(x), C1*exp(-9*x) - 4*C2*exp(-4*x)/5), Eq(g(x), C2*exp(-4*x)), Eq(h(x), (C3 + Integral(1, x))*exp(x))]
```
That factor of `Integral(0, x)` might through things off.

Reply:
It changes form depending on whether Matrix or SparseMatrix is used to send back the results from `linear_eq_to_matrix`. Simply wrapping Matrix around `SparseMatrix(*shape, dat)` will cause the results to change. This is the minimal change necessary to toggle between the two results (but it is not the root cause).